metricbeat/module/nats/varz/_meta/fields.yml

```diff
@@ -0,0 +1,77 @@
+- name: varz
+  type: group
+  description: >
+    Contains nats var related metrics
+  fields:
+    - name: server_id
+      type: keyword
+      description: >
+        The server ID
+    - name: now
+      type: date
+      description: >
+        Server time of metric creation
+    - name: uptime
+      type: long
+      description: >
+        The period the server is up
+    - name: mem
```

Comment:
Same here, it lacks the format. Maybe it's `{"mem":{"bytes":100}}`

Reply:
Could you elaborate more on this?
Nats returns mem as int64 as you can see [here](https://github.com/nats-io/gnatsd/blob/master/server/monitor.go#L841).

Reply:
This is your current data definition:

```go
	statsSchema = s.Schema{
		"server_id":         c.Str("server_id"),
		"now":               c.Str("now"),
		"uptime":            c.Str("uptime"),
		"mem":               c.Int("mem"),
	}
...
```

What I'm asking is to have something like this:

```go
	statsSchema = s.Schema{
		"server_id": c.Str("server_id"),
		"now":       c.Str("now"),
		"uptime": "sec": c.Str("uptime"),
		"mem": s.Object{
			"bytes": c.Int("mem"),
		},
...
```

In the example, `mem` has a subobject to define its format. So, if as a user I see a `mem` of _1048576_ in Elasticsearch I'll guess that they are bytes because it's too big number to be megabytes. But if the `mem` is _512_ well... maybe they are megabytes or they are actually bytes...

Of course, I can look at the docs, but our aim is to keep things simple for the user.

And yeah, that will produce a JSON line that look like `{"mem":{"bytes":100}}`. Maybe I explained it too briefly :smile: